src/timeout.rs

```diff
@@ -56,6 +56,21 @@ impl From<i32> for Timeout {
     }
 }
 
+impl From<Duration> for Timeout {
+    fn from(duration: Duration) -> Timeout {
+        if duration.is_zero() {
+            Timeout::Never
+        } else {
+            match duration.as_millis().cmp(&(u32::MAX as u128)) {
```

Comment:
I kinda have to look up what `.cmp` does now, which one is `Less` ?
Could we please go with a simple `if duration.as_millis() > u32::MAX`?

Reply:
Ok, change made.